pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -152,6 +153,15 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 		return errors.Wrapf(err, "Failed to sync to component with name %s", a.ComponentName)
 	}
 
+	// PostStart events from the devfile will only be executed when the component
+	// didn't previously exist
+	if !componentExists {
+		err = a.execDevfileEvent(a.Devfile.Data.GetEvents().PostStart, compInfo)
```

Comment:
how you are sure to execute event in container got from ` getFirstContainerWithSourceVolume`? IMO event should be executed in containerName we get from command definition.

Got it, you are updating container name below, but it looks confusing better to build compInfo in `execDevfileEvents`

Reply:
Yes. I'll pass the podName to `execDevfileEvents` instead of the compInfo.